### PR TITLE
fix for conjugate factor in mul.py

### LINKED ISSUE #16500: Assumptions: (cot(x)*cot(conjugate(x) should be real
```julia
In [43]: expr = cot(x)*cot(conjugate(x))                                                                                                       

In [47]: expr                                                                                                                                  
Out[47]: 
          ⎛_⎞
cot(x)⋅cot⎝x⎠

In [48]: print(expr.is_real)                                                                                                                   
None
```
However I think that `cot(conjugate(x))` is the same as `conjugate(cot(x))`
```julia
In [57]: cot(conjugate(x)) - conjugate(cot(x))                                                                                                 
Out[57]: 0
```
So `cot(x)*cot(conjugate(x)) = cot(x)*conjugate(cot(x)) = abs(cot(x))**2` and should be real.

### PR DESCRIPTION
**Pull request by mohitacecode**
#### References to other Issues or PRs

Fixes #16500
#### Brief description of what is fixed or changed
Now conjugated factors will get cancelled 
#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/core/mul.py [hunk 1]
```diff
@@ -1290,17 +1292,21 @@ def _eval_real_imag(self, real):
...
                 # symbolic or literal like `2 + I` or symbolic imaginary
                 if t_not_re_im:
                     return  # complex terms might cancel
                 t_not_re_im = t
-            elif t.is_imaginary is False:  # symbolic like `2` or `2 + I`
-                if t_not_re_im:
-                    return  # complex terms might cancel
-                t_not_re_im = t
             else:
                 return
```
Comment (namannimmo10) [inline sympy/core/mul.py]: That line should be removed.

Comment (namannimmo10) [inline sympy/core/mul.py]: In the `else` block, you should remove `return`.

**Comment by namannimmo10** [comment]
@mohitacecode, Please add a test.

**Comment by mohitacecode** [comment]
Why travis is failing? I can see the error but don't know how to fix it?

**Pull request closed without merging**